tests/primer/test_primer_external.py

```diff
@@ -0,0 +1,101 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+import logging
+import subprocess
+from pathlib import Path
+
+import pytest
+from pytest import LogCaptureFixture
+
+from pylint.testutils.primer import PackageToLint
+
+PRIMER_DIRECTORY = Path(".pylint_primer_tests/").resolve()
+PACKAGES_TO_LINT = {
+    "scikit-learn": PackageToLint(
+        url="https://github.com/scikit-learn/scikit-learn.git",
+        branch="main",
+        directories="sklearn",
+    ),
+    "flask": PackageToLint(
+        url="https://github.com/pallets/flask.git",
+        branch="main",
+        directories="src",
+    ),
+    "keras": PackageToLint(
+        url="https://github.com/keras-team/keras.git",
+        branch="master",
+        directories="keras",
+    ),
+    "sentry": PackageToLint(
+        url="https://github.com/getsentry/sentry.git",
+        branch="master",
+        directories="src tests",
+    ),
+    "django": PackageToLint(
+        url="https://github.com/django/django.git",
+        branch="main",
+        directories="django tests",
+    ),
+    "pandas": PackageToLint(
+        url="https://github.com/pandas-dev/pandas.git",
+        branch="master",
+        directories="pandas",
+    ),
+    "black": PackageToLint(
+        url="https://github.com/psf/black.git",
+        branch="main",
+        directories="src tests",
+    ),
+    "home-assistant": PackageToLint(
+        url="https://github.com/home-assistant/core.git",
+        branch="dev",
+        directories="homeassistant",
+    ),
+    "graph-explorer": PackageToLint(
+        url="https://github.com/vimeo/graph-explorer.git",
+        branch="master",
+        directories="graph_explorer",
+        pylintrc_relpath=".pylintrc",
+    ),
+    "pygame": PackageToLint(
+        url="https://github.com/pygame/pygame.git",
+        branch="main",
+        directories="src_py",
+    ),
+}
```

Comment:
We might want to start collecting this in a `json`, `black` does this as well.
It is the reason why I initially kept this in a separate file. Adding packages will inflate this file.

Reply:
OK let's do this :) !